### ISSUE TITLE
Distributed Training with CNTK for Python, on an Azure NC24v3 VM

### ISSUE THREAD

**Issue by JoshuaSv2** [Microsoft/CNTK#3570]
Our solution design is as follows
•	Implemented a python script, training.py, that generates and trains an RNN to map sequences to labels. The script runs successfully on a single GPU, with a standard learner, user defined training loop,  and device configuration as shown below:

#Setting default device
from cntk.device import try_set_default_device, gpu
try_set_default_device(gpu(0))
…
Learner = C.learners.fsadagrad(…)
…
for epoch in range(0, EPOCHS):
    for x_batch, l_batch in next_batch(product_sequences, product_labels):
        trainer.train_minibatch({x: x_batch, l: l_batch})

Note that next_batch is a python generator function that yields minibatches of sequences and labels contained in x_batch and l_batch respectively. To be more exact, each iteration of x_batch is a list of numpy arrays of variable row count (corresponding to the varying input sequence lengths) and each iteration of l_batch is a numpy array whose number of rows equals the number of numpy arrays in x_batch (each row in l_batch corresponds to an accompanying label) These are stored in RAM beforehand in the product_sequences and product_labels lists. 

As per the distributed training documentation, we have tried adapting the code for parallelization by using a distributed learner, MinibatchSourceFromData class and TrainingSession class as follows:

distributed_learner = distributed.data_parallel_distributed_learner(learner = learner … )

data_source = C.io.MinibatchSourceFromData(dict(x=(product_sequences, cntk.layers.typing.Sequence    [cntk.layers.typing.tensor]), l = product_labels), max_samples = 1000)
 
session = C.training_session(trainer = trainer, mb_source = data_source, mb_size = 100,
                                 model_inputs_to_streams = {x:data_source.streams['x'],
                                                            l:data_source.streams['l']}, max_samples = 1000)

However, this fails with an error:

           
![error](https://user-images.githubusercontent.com/47473021/52519134-c826c000-2cbb-11e9-8ddb-50a86058ec17.png)

We need technical guidance as to how to achieve parallelization using the TrainingSession and MinibtachSourceFromData classes (with a distributed learner).
